packages/node/src/behaviors/general-diagnostics/GeneralDiagnosticsServer.ts

```diff
@@ -103,8 +103,11 @@ export class GeneralDiagnosticsServer extends Base {
     override timeSnapshot() {
         const time = Time.nowMs();
         return {
-            systemTimeMs: time - this.internal.bootUpTime,
-            posixTimeMs: time,
+            systemTimeMs: time - Time.startup.systemMs,
+
+            // TC_DGGEN_2_4.py fails us if we set this without TimeSynchronizationCluster support.  Spec is worded
+            // poorly but my read of "SHALL only if" is not the same as "SHALL if".  But conforming to tests for now
```

Comment:
In fact yes One spec sentence is wired but in the attribute description it is described clear:
<img width="948" alt="Bildschirmfoto 2024-12-11 um 14 00 34" src="https://github.com/user-attachments/assets/61c285b1-1bf9-4004-8dc2-b04fd405ded6">

Why you do not check for existence of the TimeSync cluster and populate the value correctly as defined?

Reply:
That sentence is the one that confuses me... "shall only be null if" means “may not be null unless", not "must be null if".

So it MAY be null but is not required to be.  And why would they prevent you from providing the value just because it can't be changed?

I didn't add check for time sync because it didn't occur to me, I figured would be changed if time sync added. I'll add

Reply:
OK I added the test for TimeSynchronizationCluster.  I left the comment in but elaborated a bit more.  I still think the test suite is wrong based on formal interpretation of that language.  So at very least it's a spec issue.

Reply:
lets merge that way because seems chip understood it that way. So lets see